components/camel-swagger-java/src/main/java/org/apache/camel/swagger/servlet/RestSwaggerServlet.java

```diff
@@ -209,7 +209,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                 if (!match) {
                     adapter.noContent();
                 } else {
-                    support.renderResourceListing(adapter, swaggerConfig, name, route, json, yaml, classResolver, new RestConfiguration());
+                    support.renderResourceListing(adapter, swaggerConfig, name, route, json, yaml, null, classResolver, new RestConfiguration());
```

Comment:
I think this would lead to `NullPointerException` in `RestSwaggerSupport::setupXForwardedHeaders`. We do have the `HttpServletRequest` here so we could pass HTTP headers as a `Map<String, String>` to `support.renderResourceListing(...)`.

Reply:
That makes sense. I will change that.

Reply:
I just made Map<String, Object> to match the exchange.getIn().getHeaders, which returns <String, Object> instead of <String, String>.

Reply:
I think you missed a call to `getHeaders(request)` here